### DISCUSSION TITLE
What is the '#--unocss--xxx' selector when building Astro static?

### DISCUSSION THREAD

**Discussion by dungsil** [unocss/unocss#4233] [Q&A] [answered]
Hello, I am creating a project using Astro v4 (static) + UnoCSS.
When I build a simple integration, I see a special selector like the one below. Why is it created and what does it do?

```html
<!-- dist/index.html -->
<!-- ... -->
  <style>
    #--unocss-- {
      layer: __ALL__;
      escape-view: \"\'\`\\
    }

    #--unocss-layer-start--__ALL__-- {
      start: __ALL__
    }

  /* ... */

    #--unocss-layer-end--__ALL__-- {
      end: __ALL__
    }
</style>
```

+)
On the development server, you will see /* layer: preflights */ instead of the selector below. And the built output doesn't seem to include the unocss-related runtime. Could this be a part created incorrectly during the SSG process?

**Comment by sibbng** [comment, marked as answer]
These are placeholders that is supposed to be replaced during build. I guess it didn't work in that case because styles are inlined in the index.html. Can you try disabling the following option in Astro config: 

https://docs.astro.build/en/reference/configuration-reference/#buildinlinestylesheets

**Reply by dungsil** [reply]
Thanks, it's solved!
However, is there any way to handle it inline?

**Reply by sibbng** [reply]
Not yet 😔
